packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -220,9 +220,13 @@ export class OdspDelayLoadedDeltaStream {
 				this.currentConnection = connection;
 				return connection;
 			} catch (error) {
-				this.clearJoinSessionTimer();
-				this.cache.sessionJoinCache.remove(this.joinSessionKey);
-
+				// Remove join session information from cache only if it is an error related to connect_document and not a socket related error.
+				// Otherwise keep it in cache so that this session can be re-used after disconnection.
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+				if ((error as any).errorFrom === "connect_document_error") {
```

Comment:
I wonder if I should (as a fallback) also add a check for undefined scenario - clear cache if errorFrom is not available in the error property and also log it as an error, so that we can revisit the error stack and see which code path caused it. cc @jatgarg / @agarwal-navin

Reply:
Yes, you can do that to retain the original logic but I would suggest not to make an error event but just a normal telemetry event and you can track it later on as error event raises eyebrows for other people who sees it in telemetry. This event would be more for you to see whether there was a bug in your logic?

Reply:
Decided to keep the undefined check, but not log it. I realized the cost of adding, monitoring and then removing the log for undefined code path log is to high than the benefit we will get out of it.